devito/builtins.py

```diff
@@ -205,6 +205,29 @@ def fset(f, g):
     return f
 
 
+def nbl_to_padsize(nbl, ndim):
+    """
+    Creates the pad sizes from `nbl`. The output is a tuple of tuple
+    such as ((40, 40), (40, 40)) where each subtuple is the left/right
+    pad size for the dimension
+    """
+    nb_total = as_tuple(nbl)
+    if len(nb_total) == 1 and len(nb_total) < ndim:
+        nb_pad = ndim*((nb_total[0], nb_total[0]),)
+    elif len(nb_total) == ndim:
+        nb_pad = []
+        for nb_dim in nb_total:
+            if len(as_tuple(nb_dim)) == 1:
+                nb_pad.append((nb_dim, nb_dim))
+            else:
+                assert len(nb_dim) == 2
+                nb_pad.append(nb_dim)
+    else:
+        raise ValueError("nbl must be an integer or tuple of integers of length" +
```

Comment:
quotes around nbl and around function.shape

also, no need for '+'

Reply:
fixed